Error in find_translation_path

@jim22k, I *sometimes* get this failure when running tests on master:
```python-traceback
_________________________________________________________ test_python _________________________________________________________

default_plugin_resolver = <metagraph.core.resolver.Resolver object at 0xa1c9e6550>

    def test_python(default_plugin_resolver):
        dpr = default_plugin_resolver
        sparse_dict = {1: 1.1, 5: 5.5, 2: 2.2}
        x = PythonSparseVector(sparse_dict, size=30)
        assert len(x) == 30
        # Convert back and forth from numpy sparse array
        nps = dpr.translate(x, NumpySparseVector)
        assert isinstance(nps, NumpySparseVector)
>       y = dpr.translate(nps, PythonSparseVector)

metagraph/tests/translators/test_sparsevector.py:15:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
metagraph/core/resolver.py:338: in translate
    translator = self.find_translator(value, dst_type)
metagraph/core/resolver.py:325: in find_translator
    ret_val = self.find_translation_path(src_type, dst_type)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <metagraph.core.resolver.Resolver object at 0xa1c9e6550>
src_type = <class 'metagraph.default_plugins.wrappers.numpy.NumpyVectorType'>
dst_type = <class 'metagraph.default_plugins.wrappers.python.PythonSparseVectorType'>

    def find_translation_path(self, src_type, dst_type) -> Optional[Translator]:
        import scipy.sparse as ss

        abstract = dst_type.abstract
        if abstract not in self.translation_matrices:
            # Build translation matrix
            concrete_types = [
                ct for ct in self.concrete_types if issubclass(ct.abstract, abstract)
            ]
            m = ss.dok_matrix((len(concrete_types), len(concrete_types)), dtype=bool)
            for s, d in self.translators:
                try:
                    sidx = concrete_types.index(s)
                    didx = concrete_types.index(d)
                    m[sidx, didx] = True
                except ValueError:
                    pass
            sssp, predecessors = ss.csgraph.dijkstra(
                m.tocsr(), return_predecessors=True, unweighted=True
            )
            self.translation_matrices[abstract] = (concrete_types, sssp, predecessors)
        # Lookup shortest path from stored results
        concrete_types, sssp, predecessors = self.translation_matrices[abstract]
>       sidx = concrete_types.index(src_type)
E       ValueError: <class 'metagraph.default_plugins.wrappers.numpy.NumpyVectorType'> is not in list
```

--- a/metagraph/core/resolver.py
+++ b/metagraph/core/resolver.py
@@ -76,9 +76,10 @@ class Resolver:
 
         # translation graph matrices
         # Single-sourch shortest path matrix and predecessor matrix from scipy.sparse.csgraph.dijkstra
-        # Stored result is (concrete_types list, sssp_matrix, predecessors_matrix)
+        # Stored result is (concrete_types list, concrete_types lookup, sssp_matrix, predecessors_matrix)
         self.translation_matrices: Dict[
-            AbstractType, Tuple[List[ConcreteType], np.ndarray, np.ndarray]
+            AbstractType,
+            Tuple[List[ConcreteType], Dict[ConcreteType, int], np.ndarray, np.ndarray],
         ] = {}
 
         self.algo = Namespace()
@@ -281,27 +282,38 @@ class Resolver:
         abstract = dst_type.abstract
         if abstract not in self.translation_matrices:
             # Build translation matrix
-            concrete_types = [
-                ct for ct in self.concrete_types if issubclass(ct.abstract, abstract)
-            ]
-            m = ss.dok_matrix((len(concrete_types), len(concrete_types)), dtype=bool)
+            concrete_list = []
+            concrete_lookup = {}
+            for ct in self.concrete_types:
+                if issubclass(ct.abstract, abstract):
+                    concrete_lookup[ct] = len(concrete_list)
+                    concrete_list.append(ct)
+            m = ss.dok_matrix((len(concrete_list), len(concrete_list)), dtype=bool)
             for s, d in self.translators:
-                try:
-                    sidx = concrete_types.index(s)
-                    didx = concrete_types.index(d)
-                    m[sidx, didx] = True
-                except ValueError:
-                    pass
+                # only accept destinations of specific abstract type
+                if d.abstract == abstract:
+                    try:
+                        sidx = concrete_lookup[s]
+                        didx = concrete_lookup[d]
+                        m[sidx, didx] = True
+                    except KeyError:
+                        pass
             sssp, predecessors = ss.csgraph.dijkstra(
                 m.tocsr(), return_predecessors=True, unweighted=True
             )
-            self.translation_matrices[abstract] = (concrete_types, sssp, predecessors)
+            self.translation_matrices[abstract] = (
+                concrete_list,
+                concrete_lookup,
+                sssp,
+                predecessors,
+            )
         # Lookup shortest path from stored results
-        concrete_types, sssp, predecessors = self.translation_matrices[abstract]
+        packed_data = self.translation_matrices[abstract]
+        concrete_list, concrete_lookup, sssp, predecessors = packed_data
         try:
-            sidx = concrete_types.index(src_type)
-            didx = concrete_types.index(dst_type)
-        except ValueError:
+            sidx = concrete_lookup[src_type]
+            didx = concrete_lookup[dst_type]
+        except KeyError:
             return None
         if sssp[sidx, didx] == np.inf:
             return None
@@ -310,7 +322,7 @@ class Resolver:
         while True:
             parent_idx = predecessors[sidx, didx]
             steps.insert(
-                0, self.translators[(concrete_types[parent_idx], concrete_types[didx])]
+                0, self.translators[(concrete_list[parent_idx], concrete_list[didx])]
             )
             if parent_idx == sidx:
                 break

